### DISCUSSION TITLE
Possible Issue during initial embedding stage

### DISCUSSION THREAD

**Discussion by abhuva** [brianpetro/obsidian-smart-connections#360] [General]
I am using this plugin for quite a while now - so far with moderate size vaults (around 1000 notes).
Now i tried to use it on my ttrpg gaming vault (10k notes, around 6 million words)
So far the initial embedding api calls did cost me 12$ - this is a huge amount of requests for embeddings.
Also, the stored embeddings file doesnt grow (i was expecting it to save the embeddings there, but the last change in this file is several hours ago, while the plugin keeps requesting api calls)

Obsidian crashed inbetween a couple times (exceeding 5GB Ram usage)

Now i wonder:
1) is the progress (the downloaded embeddings) stored inbetween
2) can i close obsidian while the process isnt finished - and it will pickup where it left once i restart?
3) is there any way to see how many notes are already processed?

**Comment by brianpetro** [comment]
@abhuva unfortunately, I think this is an unresolved issue, see https://github.com/brianpetro/obsidian-smart-connections/issues/356#issuecomment-1782133476 for more info.

🌴

**Reply by abhuva** [reply]
I did read through the comments - one thing that is different: i didnt see any errors in the dev console. Only 2 things:
batch embeddings and scheduled save

I did setup the "static" part of my ttrpg vault new (this time 1k file around 1 million words) - forced a refresh and this time it finishes the process. Embedding file size around 250mb

My backup (failed) embedding size was 500mb.
So yes, after a certain vault size this breaks...

**Reply by brianpetro** [reply]
Good to know. The size is a factor of the number of embeddings, not so much the number of words, so to reduce the total embeddings you could try the `skip_sections` setting, which might allow you to gain some more coverage over your notes. I can't say for certain that it will work, though (still might be too big).

Thanks for the follow-up. And the size-issue should be fixed in the next major update.

🌴
